soc/arm/nxp_imx/rt/soc_rt10xx.c

```diff
@@ -224,9 +224,16 @@ static ALWAYS_INLINE void clock_init(void)
 	defined(CONFIG_MEMC_MCUX_FLEXSPI) && \
 	DT_NODE_HAS_STATUS(DT_NODELABEL(flexspi), okay)
 	CLOCK_DisableClock(kCLOCK_FlexSpi);
-	CLOCK_InitUsb1Pfd(kCLOCK_Pfd0, 24);
-	CLOCK_SetMux(kCLOCK_FlexspiMux, 3);
-	CLOCK_SetDiv(kCLOCK_FlexspiDiv, 2);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexspi), okay) && \
+	defined(CONFIG_MEMC_MCUX_FLEXSPI)
```

Comment:
We should be adjusting the FlexSpi clocks only in cases when we do not `xip` from Flash. Your change removes this check.

Reply:
That is true. However without changing these clocks the imxrt1050_evk hyperflash does not work. Using the sample to write:
``` flash page_write 100 0 1 2 3 4 ```
reads back 1 6 3 4.
I have an idea that the dcd file for imxrt1050 might be wrong.
Should I instead create a bug-report?

Reply:
It might be worth revisiting the LUT table settings or commands used in the Flash driver